lib/fastlane_core/pkg_file_analyser.rb

```diff
@@ -0,0 +1,39 @@
+require 'nokogiri'
+
+module FastlaneCore
+  class PkgFileAnalyser
+    def self.fetch_app_identifier(path)
+      xml = self.fetch_distribution_xml_file(path)
+      return xml.at_css('product')['id'] if xml
+      return nil
+    end
+
+    # Fetches the app version from the given pkg file.
+    def self.fetch_app_version(path)
+      xml = self.fetch_distribution_xml_file(path)
+      return xml.at_css('product')['version'] if xml
+      return nil
+    end
+
+    def self.fetch_distribution_xml_file(path)
+      Dir.mktmpdir do |dir|
+        `xar -C #{dir} -xf #{path}`
+
+        Dir.foreach(dir) do |file|
+          next unless file.include? 'Distribution'
+
+          begin
+            xml = File.open(File.join(dir, file)) { |f| Nokogiri::XML(f) }
+            if xml.at_css('product')
+              return xml
+            end
+          rescue
```

Comment:
Maybe this should be a specific exception that we catch, instead of everyone, in case we get another error